### PR TITLE
:mortar_board: Asn1 `CountryCatalogue` Second Half

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
Follows #414 

### What
Add more questions/functionality for/to the class

### Where
This will follow the first half of questions in #414. Note the numbering will start at `1` in this series of questions as a consequence of using `#`. Once combined with #414, the numbering will be correct. 

### Testing
Built local

### Additional Notes
1. I was not a huge fan of the spacing for the `CountryCatalogue`'s `toString` example. I tried to make it a code block, but then it started emphazising things weird, so I stuck with the verbatum/codelike text. 
2. I originally wanted to have ``filterCountriesByContinent`` return a new ``CountryCatalogue``, but thought that perhaps that would be a _little much_ at this stage --- "wait, objects can create instances of themselves? :exploding_head:" Changing this to be a new ``CountryCatalogue`` would almost be simpler, but it would require another constructor. Either way, the array version has them keeping track of seperate indicies for the filtered catalogue and the catalogue.

### CODE REVIEW

File: src/site/asn1.rst [hunk 1]
```diff
@@ -69,6 +69,36 @@ Part 2 --- Country Catalogue
 ============================
 
 
+#. Create a method called ``setPopulation`` that takes a name of a country and a new population value for that country as parameters
+
+#. Have a method called ``numberCountriesOnContinent`` that takes a continent name as a parameter and returns the number of ``Country`` objects on that continent
+
+#. Write a method called ``filterCountriesByContinent`` that takes a continent name as a parameter and returns a ``Country`` array containing only the countries on the specified continent
+    * You may find ``numberCountriesOnContinent`` useful here
```
Comment (jameshughes89) [inline src/site/asn1.rst:77]: I was considering this to be returning a new ``CountryCatalogue`` as opposed to an array of ``Country`` objects.

File: src/site/asn1.rst [hunk 2]
```diff
@@ -69,6 +69,36 @@ Part 2 --- Country Catalogue
...
+
+#. Write the ``toString`` method to return a ``String`` with each ``Country`` object's details on a separate line
+    * The ``Country`` class' ``toString`` will be useful for getting the object's details
+    * Below is an example
+
+            ``China, Asia, 1339190000, 9596960.0``
+
+            ``United States of America, North America, 309975000, 9629091.0``
+
+            ``Brazil, South America, 193364000, 8511965.0``
```
Comment (jameshughes89) [inline src/site/asn1.rst:91]: I do not love the spacing here or in the rendered webpage, but whatever.

File: src/site/asn1.rst (hunk 1 above)
Comment (twentylemon) [reply src/site/asn1.rst:77]: An array constructor is simple enough I think to add it. Implementation is just `for each country { add(country) }`. Such a constructor would make the signature a little more sane, and not really be any harder imo.

File: src/site/asn1.rst [hunk 3]
```diff
@@ -69,6 +69,36 @@ Part 2 --- Country Catalogue
...
+
+
+#. Write a ``saveCatalogue`` method to save the string representing the ``CountryCatalogue`` to a specified file
```
Comment (twentylemon) [inline src/site/asn1.rst:94]: This again seems like it belongs outside of the class itself., related to #414 comments.

Comment (jameshughes89) [reply src/site/asn1.rst:94]: Upon further thought, this will just get cut.

File: src/site/asn1.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/asn1.rst:77]: Given the reworking of the constructor in #414, this will be trivial to adjust now.

File: src/site/asn1.rst [hunk 4]
```diff
@@ -69,6 +69,28 @@ Part 2 --- Country Catalogue
 ============================
 
 
+#. Create a method called ``setPopulation`` that takes a name of a country and a new population value for that country as parameters
+
+#. Have a method called ``numberCountriesOnContinent`` that takes a continent name as a parameter and returns the number of ``Country`` objects on that continent
+
+#. Write a method called ``filterCountriesByContinent`` that takes a continent name as a parameter and returns a new ``CountryCatalogue`` instance containing only the countries on the specified continent
```
Comment (jameshughes89) [inline src/site/asn1.rst:76]: I was gonna' have a note about having an instance of `CountryCatalogue` making a new instance of a `CountryCatalogue`, but I opted to not bother.

File: src/site/asn1.rst (hunk 3 above)
Comment (jameshughes89) [reply src/site/asn1.rst]: :hocho:

**Review by twentylemon — APPROVED** [review]

**Pull request merged**